libc-top-half/musl/src/thread/pthread_create.c

```diff
@@ -164,14 +164,6 @@ static void __pthread_exit(void *result)
 	self->prev->next = self->next;
 	self->prev = self->next = self;
 
-#ifndef __wasilibc_unmodified_upstream
-	/* On Linux, the thread is created with CLONE_CHILD_CLEARTID,
-	 * and this lock will unlock by kernel when this thread terminates.
-	 * So we should unlock it here in WebAssembly.
-	 * See also set_tid_address(2) */
-	__tl_unlock();
-#endif
```

Comment:
Shouldn't we be leaving this in since presumably it is the original code and `__wasilibc_unmodified_upstream` is intended to mark that?

Reply:
no.
i guess you are confused between `#ifdef` and `#ifndef`.

Reply:
Oh, sorry!